python/ml4ir/applications/ranking/model/ranking_model.py

```diff
@@ -121,167 +121,175 @@ def evaluate(
         eval_dict = prepare_eval_config_for_ranking(self.eval_config, group_keys)
         group_keys = eval_dict[EvalConfigConstants.GROUP_BY]
 
-        metrics_dict = dict()
-        group_metrics_keys = self.feature_config.get_group_metrics_keys()
-        evaluation_features = (
-            group_metrics_keys
-            + [
-                self.feature_config.get_query_key(),
-                self.feature_config.get_label(),
-                self.feature_config.get_rank(),
-            ]
-        )
-
-        # Add aux_label if present
-        aux_label = self.feature_config.get_aux_label()
-        if aux_label and (
-                aux_label.get("node_name") or (
-                aux_label["name"] not in eval_dict[EvalConfigConstants.GROUP_BY])):
-            evaluation_features += [aux_label]
-
-        additional_features[metrics_helper.RankingConstants.NEW_RANK] = prediction_helper.convert_score_to_rank
+        # If basic mode is specified, only compute the keras Model metrics
+        # By default, use the extended mode and compute metrics as defined in this function
+        if eval_dict.get(EvalConfigConstants.MODE, EvalConfigConstants.EXTENDED_MODE) == EvalConfigConstants.BASIC_MODE:
+            metrics_dict = self.model.evaluate(test_dataset, return_dict=True)
+            metrics_dict = {f"test_{key}": val for key, val in metrics_dict.items()}
+            self.logger.info("Overall Metrics: \n{}".format(pd.Series(metrics_dict)))
+            return None, None, metrics_dict
```

Comment:
@mohazahran This is the main change. The rest is indentation due to `if-else` block.